### PR TITLE
Onboard IaaS volume commands

### PR DESCRIPTION
**Pull request by GokceGK**
Volume

- list
- describe
- create
- delete
- resize
- update
- performance-class
       - list
       - describe

### CODE REVIEW

File: internal/cmd/beta/volume/create/create.go [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+				`$ stackit beta volume create --availability-zone eu01-1`,
+			),
+			examples.NewExample(
+				`Create a volume with name "volume-1"`,
+				`$ stackit beta volume create --availability-zone eu01-1 --name volume-1`,
+			),
+			examples.NewExample(
+				`Create a volume with availability zone "eu01-1", performance class "storage_premium_perf1"`,
+				`$ stackit beta volume create --availability-zone eu01-1 --performance-class storage_premium_perf1`,
+			),
```
[the remark this answers is not in the document]
Comment (joaopalet) [reply internal/cmd/beta/volume/create/create.go:66]: Either way, we should have an example to create a volume given a size, at least one from an image for example

File: internal/cmd/beta/volume/describe/describe.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+		table.AddRow("VOLUME SIZE (GB)", *volume.Size)
+		table.AddSeparator()
+		table.AddRow("PERFORMANCE CLASS", *volume.PerformanceClass)
+		table.AddSeparator()
+		table.AddRow("AVAILABILITY ZONE", *volume.AvailabilityZone)
+		table.AddSeparator()
+		table.AddRow("SOURCE", sourceId)
+		table.AddSeparator()
+		table.AddRow("SERVER", serverId)
+		table.AddSeparator()
```
Comment (joaopalet) [inline internal/cmd/beta/volume/describe/describe.go:148]: This will show these field even if they're empty, right? We can show conditionally

Comment (joaopalet) [reply internal/cmd/beta/volume/describe/describe.go:148]: Also we could show the size, if the field is present

File: internal/cmd/beta/volume/list/list.go [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+		Args:  args.NoArgs,
+		Example: examples.Build(
+			examples.NewExample(
+				`Lists all volumes`,
+				"$ stackit beta volume list",
+			),
+			examples.NewExample(
+				`Lists all volumes which contains the label xxx`,
+				"$ stackit beta volume list --label-selector xxx",
+			),
```
Comment (joaopalet) [inline internal/cmd/beta/volume/list/list.go:48]: 👍

File: internal/cmd/beta/volume/list/list.go [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+		details, err := yaml.MarshalWithOptions(volumes, yaml.IndentSequence(true))
+		if err != nil {
+			return fmt.Errorf("marshal volume: %w", err)
+		}
+		p.Outputln(string(details))
+
+		return nil
+	default:
+		table := tables.NewTable()
+		table.SetHeader("ID", "Name", "Status", "Server", "Public IP")
```
Comment (joaopalet) [inline internal/cmd/beta/volume/list/list.go:167]: The header columns are not matching the row values

File: internal/cmd/beta/volume/update/update.go [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+	payload := iaas.UpdateVolumePayload{
+		Name:        model.Name,
+		Description: model.Description,
+		Labels:      labelsMap,
+	}
+
+	return req.UpdateVolumePayload(payload)
+}
+
+func outputResult(p *print.Printer, model *inputModel, projectLabel string, volume *iaas.Volume) error {
```
Comment (joaopalet) [inline internal/cmd/beta/volume/update/update.go:151]: From the `RunE`, we are passing the volume name as third argument (and not the project label)

Comment (joaopalet) [reply internal/cmd/beta/volume/update/update.go:151]: The output text in `default:` needs to be adjusted also

File: internal/cmd/beta/volume/resize/resize.go [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+			p.Outputf("Resized volume %q.\n", volumeLabel)
+			return nil
+		},
+	}
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().Int64(sizeFlag, 0, "Volume size")
```
Comment (joaopalet) [inline internal/cmd/beta/volume/resize/resize.go:88]:
```suggestion
	cmd.Flags().Int64(sizeFlag, 0, "Volume size (GB)")
```

File: internal/cmd/beta/volume/describe/describe.go (hunk 2 above)
Comment (GokceGK) [reply internal/cmd/beta/volume/describe/describe.go:148]: I think volume will always have a size even if we dont parse it in creation

File: internal/cmd/beta/volume/create/create.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (GokceGK) [reply internal/cmd/beta/volume/create/create.go:66]: nice catch, thx. Edited the examples to cover both cases

**Review by joaopalet — APPROVED** [review]

**Pull request merged**